tests/test_new_types.py

```diff
@@ -0,0 +1,125 @@
+from datetime import date
+from decimal import Decimal
+from typing import List, NewType, Optional, Union
+
+from pydantic import (
+    BaseModel,
+    PositiveFloat,
+    conbytes,
+    condate,
+    condecimal,
+    confloat,
+    confrozenset,
+    conint,
+    conlist,
+    conset,
+    constr,
+)
+
+from pydantic_factories import ModelFactory
+
+
+def test_new_types() -> None:
+    MyInt = NewType("MyInt", int)
+    WrappedInt = NewType("WrappedInt", MyInt)
+
+    class MyModel(BaseModel):
+        int_field: MyInt
+        wrapped_int_field: WrappedInt
+
+    class MyModelFactory(ModelFactory):
+        __model__ = MyModel
+
+    result = MyModelFactory.build()
+    assert isinstance(result.int_field, int)
+    assert isinstance(result.wrapped_int_field, int)
+
+
+def test_complex_new_types() -> None:
+    MyStr = NewType("MyStr", str)
+    MyInt = NewType("MyInt", int)
+
+    class NestedModel(BaseModel):
+        nested_int_field: MyInt
+
+    MyNestedModel = NewType("MyNestedModel", NestedModel)
+
+    class MyModel(BaseModel):
+        list_int_field: List[MyInt]
+        union_field: Union[MyInt, MyStr]
+        optional_str_field: Optional[MyStr]
+        nested_model_field: MyNestedModel
+
+    class MyModelFactory(ModelFactory):
+        __model__ = MyModel
+
+    result = MyModelFactory.build()
+
+    assert isinstance(result.union_field, (int, str))
+    assert result.optional_str_field is None or isinstance(result.optional_str_field, str)
+    assert isinstance(result.nested_model_field, NestedModel)
+    assert isinstance(result.nested_model_field.nested_int_field, int)
+    assert isinstance(result.list_int_field[0], int)
+
+
+def test_constrained_new_types() -> None:
+    ConBytes = NewType("ConBytes", conbytes(min_length=2, max_length=4))  # type: ignore[misc]
```

Comment:
Here and below mypy complains that `conbytes(min_length=2, max_length=4)` is not a valid type